assets/src/design-system/components/radio/radio.js

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { forwardRef, useMemo } from 'react';
+import styled, { css } from 'styled-components';
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * Internal dependencies
+ */
+import { themeHelpers, THEME_CONSTANTS } from '../../theme';
+import { FOCUS_VISIBLE_SELECTOR } from '../../theme/global';
+import { Text } from '../typography';
+
+const BORDER_WIDTH = 1;
+const TEXT_GAP_HEIGHT = 8;
+const RING_DIAMETER = 24;
+const CONTAINER_HEIGHT = 44;
+const RADIO_DIAMETER = 16;
+
+const Container = styled.div`
+  display: flex;
+  align-items: center;
+  min-height: ${CONTAINER_HEIGHT}px;
+  padding: 8px 0;
+`;
+
+const ButtonContainer = styled.div`
+  position: relative;
+  height: ${RING_DIAMETER}px;
+  width: ${RING_DIAMETER}px;
+`;
+
+const LabelContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  margin: 0 12px;
+  gap: ${TEXT_GAP_HEIGHT}px;
+`;
+
+const RadioBorder = styled.span(
+  ({ theme }) => css`
+    display: inline-block;
+    box-sizing: border-box;
+    height: ${RING_DIAMETER}px;
+    width: ${RING_DIAMETER}px;
+    border: ${BORDER_WIDTH}px solid ${theme.colors.border.defaultNormal};
+    border-radius: ${theme.borders.radius.round};
+    pointer-events: none;
+    transition: all 0.15s;
+  `
+);
+
+const InnerButton = styled.span(
+  ({ theme }) => css`
+    pointer-events: none;
+    display: none;
+
+    :after {
+      content: '';
+      position: absolute;
+      height: ${RADIO_DIAMETER}px;
+      width: ${RADIO_DIAMETER}px;
+      top: 50%;
+      left: 50%;
+      transform: translate(-50%, -50%);
+      background-color: ${theme.colors.fg.primary};
+      border-radius: ${theme.borders.radius.round};
+    }
+  `
+);
+
+const HiddenInput = styled.input.attrs({ type: 'radio' })`
+  position: absolute;
+  height: ${RING_DIAMETER}px;
+  width: ${RING_DIAMETER}px;
+  margin: 0;
+  opacity: 0;
```

Comment:
Nit: Should this use `${themeHelpers.visuallyHidden};` to do just that?

Reply:
Yeah totally! I felt like I needed something a little different here. The idea was that the input would cover all of the space of the "radio" styles (outer ring + container) so that clicking anywhere on it would click the input

Using the theme helper would make the input 1px by 1px:
```
// themeHelpers.visuallyHidden
position: absolute;
height: 1px;
width: 1px;
overflow: hidden;
clip: rect(1px, 1px, 1px, 1px);
white-space: nowrap;
```

The visually hidden styles rely on the `clip` and `overflow: hidden` but I think I was leaning more towards making the element bigger than 1px by 1px and making it invisible but still readable by screen readers.

If we used that here it would look like this still:
```
${themeHelpers.visuallyHidden};
height: ${RING_DIAMETER}px;
width: ${RING_DIAMETER}px;
clip: auto;
opacity: 0;
```

This would overwrite most of those styles anyways, so I figured I'd just add the `opacity: 0` and the width + height. But for consistency I'm happy to add it if we feel like it should be used here!

Reply:
Yea I think this is a different use case than what our visually hidden theme helper was made for. I don't see any point in using it cus it just adds more functionality you're either overwriting or don't want.